### PR TITLE
Updating tb-profiler to 6.4.0

### PR DESCRIPTION
**Pull request by emarinier**
FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] License permits unrestricted use (educational + commercial)
* [ ] This PR adds a new tool or tool collection
* [x] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

Related to https://github.com/galaxyproject/tools-iuc/pull/5139

### CODE REVIEW

**Comment by emarinier** [comment]
Testing error appears to be:

```
[E::hts_open_format] Failed to open file "vcf/tbprofiler.targets.vcf.gz" : No such file or directory
Failed to read from vcf/tbprofiler.targets.vcf.gz: No such file or directory
```

and possibly:

```
[19:57:03] INFO     Running delly                                      bam.py:84
           ERROR    delly: error while loading shared libraries:    utils.py:534
                    libboost_iostreams.so.1.85.0: cannot open                   
                    shared object file: No such file or directory               
                                                                                
           ERROR    Delly failed, skipping                             bam.py:96
```

Which does not occur when running `planemo test` locally.

**Comment by emarinier** [comment]
The solution appears to be fixing the version of delly:

```
<requirements>
    <requirement type="package" version="1.2.6">delly</requirement>
    <requirement type="package" version="@TOOL_VERSION@">tb-profiler</requirement>
</requirements>
```

**Comment by bgruening** [comment]
@pvanheus do you want to have a look?

@emarinier thanks a lot

File: tools/tb-profiler/tb_profiler_profile.xml [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
     <description>Infer strain types and drug resistance markers from sequences</description>
     <macros>
-        <token name="@TOOL_VERSION@">6.2.1</token>
+        <token name="@TOOL_VERSION@">6.4.0</token>
     </macros>
     <xrefs>
         <xref type="bio.tools">tb-profiler</xref>
     </xrefs>
     <requirements>
+        <requirement type="package" version="1.2.6">delly</requirement>
```
Comment (bgruening) [inline tools/tb-profiler/tb_profiler_profile.xml:10]: If this is true, we should maybe fix the conda package.

Comment (bgruening) [reply tools/tb-profiler/tb_profiler_profile.xml:10]: Oh delly is not even a dependency of tb-profiler. https://github.com/bioconda/bioconda-recipes/blob/master/recipes/tb-profiler/meta.yaml

It is a dependency of https://github.com/bioconda/bioconda-recipes/blob/master/recipes/pathogen-profiler/meta.yaml

Comment (bgruening) [reply tools/tb-profiler/tb_profiler_profile.xml:10]: I guess then we leave it like it is.

**Comment by bgruening** [comment]
Thanks @emarinier

File: tools/tb-profiler/tb_profiler_profile.xml (hunk 1 above)
Comment (pvanheus) [reply tools/tb-profiler/tb_profiler_profile.xml:10]: Coming to this late - pathogen-profiler is the underlying tool that tb-profiler uses. It [depends](https://github.com/bioconda/bioconda-recipes/blob/master/recipes/pathogen-profiler/meta.yaml#L34) on delly 1.2.6. The tb-profiler 4.6.0 in bioconda depends on pathogen-profiler 4.5.0 which has exactly this dependency (delly 1.2.6). I don't understand why an extra requirement is needed in the tool wrapper?

Comment (emarinier) [reply tools/tb-profiler/tb_profiler_profile.xml:10]: I wonder if it's related to Aaron's comment here: https://github.com/jodyphelan/TBProfiler/issues/371#issuecomment-2197716747 where there's two builds of Delly 1.2.6?

**Pull request merged**